### PR TITLE
Fix shellcheck for godep scripts in hack

### PR DESCRIPTION
**Pull request by mattjmcnaughton**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
Fix errors highlighted by `shellcheck` for `hack/godep-restore.sh` and
`hack/godep-save.sh`. Part of the effort tracked in https://github.com/kubernetes/kubernetes/issues/72956.

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

**Comment by mattjmcnaughton** [comment]
/assign @cblecker 

thanks :)

**Comment by mrbobbytables** [comment]
Thanks! quick look over looks good.

/ok-to-test

File: hack/godep-save.sh
Comment (spiffxp) [inline hack/godep-save.sh]: You're hitting another shellcheck failure:
```
In ./hack/godep-save.sh line 82:
  repo=$(basename ${full_repo_path})
                  ^---------------^ SC2086: Double quote to prevent globbing and word splitting.
```

More to the point, what was the original failure you were addressing, and why do it this way?

**Review by spiffxp — CHANGES_REQUESTED** [review]
need to fix a failure, and this still feels like it's vulnerable to spaces

File: hack/godep-save.sh
Comment (mattjmcnaughton) [reply hack/godep-save.sh]: Great question! I was fixing https://github.com/koalaman/shellcheck/wiki/SC2045, which says to prefer shell globs over iterating over `ls`. I'll also fixed the additional shellcheck failure - thanks for catching that :)

**Comment by spiffxp** [comment]
/hold
for a question in a different PR

**Comment by mattjmcnaughton** [comment]
@spiffxp mind LGTM'ing? Your previous lgtm got deleted when I rebased. thanks!

**Comment by mattjmcnaughton** [comment]
Closing this because Godep is no more :100:

**Pull request closed without merging**
